File: 2.Het/2. Szülő-gyerek/Szulo_Gyerek.java
class Szulo {
	void szulometodus() {
		System.out.println("Szulo vagyok!");
	}
}

class Gyerek extends Szulo {
	void gyerekmetodus() {
		System.out.println("Gyerek vagyok!");
	}
}

class Szulo_Gyerek {
	public static void main(String[] args) {
	Szulo valaki = new Gyerek();
	
	valaki.szulometodus();
	//valaki.gyerekmetodus();
	}
}

File: 2.Het/2. Szülő-gyerek/Szulo_Gyerek.cpp
#include <iostream>

class Szulo {
public:
	void szulometodus() {
		std::cout << "Szulo vagyok!" << std::endl;
	}
};

class Gyerek : public Szulo {
public:
	void gyerekmetodus() {
		std::cout << "Gyerek vagyok!" << std::endl;
	}
};

int main() {
	Szulo* valaki = new Gyerek();

	valaki->szulometodus();
	valaki->gyerekmetodus();
}